dsa-23au/java-dsa/jpract-app/src/main/java/com/practice/app/mergeSort.java

```diff
@@ -0,0 +1,75 @@
+package main.java.com.practice.app;
+
+public class mergeSort 
+{
+    public void sort(int[] inputArray)
+    {
+		
+		//Instantiating the variables length and midpoint
+		int length = inputArray.length;
+		int midpoint = length / 2;
+		
+		//setting leftSide size equal to the length
+		//up to midpoint
+		//setting rightSide equal to the remaining length
+		int[] leftSide = new int[midpoint];
+		int[] rightSide = new int[length - midpoint];
```

Comment:
This sort instantiates new arrays to represent the left and right sides of this sub-array rather than referencing them using array indices as in the in-class examples